backup/moodle2/backup_hvp_activity_task.class.php

```diff
@@ -40,9 +40,31 @@ protected function define_my_settings() {
     }
 
     /**
-     * Defines a backup step to store the instance data in the choice.xml file
+     * Defines a backup step to store the instance data in the hvp.xml file
      */
     protected function define_my_steps() {
-        // $this->add_step(new backup_hvp_activity_structure_step('hvp_structure', 'hvp.xml'));
+        $this->add_step(new backup_hvp_activity_structure_step('hvp_structure', 'hvp.xml'));
+    }
+
+    /**
+     * Code the transformations to perform in the activity in
+     * order to get transportable (encoded) links
+     *
+     * @param string $content
+     * @return string encoded content
+     */
+    static public function encode_content_links($content) {
+        // TODO Should the paths like this:
+        //     {"path":"images/imageSlideBackground-56c5a34e5743e.jpg","mime"
+        // be converted here into format:
+        //     {"path":"@@PLUGINFILE@@/image.jpg","mime"
+        // or something similar so that the backup system's annotate_files()
+        // method would be able to include them into the backup?
+
+        // This regex already matches all jpg files in such way:
+        $search = "/\"path\":\"images\/([^\"]+)\",\"mime\"/";
+        $content = preg_replace($search, '@@PLUGINFILE@@/${1}","mime"', $content);
```

Comment:
I've most likely misunderstood this part. Apparently the backup feature does not check existence of files in the text fields, but instead fetches them from the `mdl_files` table directly.